### DISCUSSION TITLE
Gsoc first issue: Beginner

### DISCUSSION THREAD

**Discussion by meesha2000** [checkstyle/checkstyle#13974] [General]
Hello,

I am new to open source contribution and would like to start by contributing to CheckStyle. I have successfully set up the project on my PC, but I'm encountering errors when trying to run it. Additionally, I'm unsure how to tackle and begin working on GSoC Good First Issues.

I'd greatly appreciate your guidance and assistance in getting started with CheckStyle. Could you please help me with the following:

Resolving Setup Errors: I have encountered errors while trying to run the project on my local machine. What steps should I take to troubleshoot and resolve these errors?

Error: The declared package "" does not match the expected package "com.puppycrawl.tools.checkstyle.checks.imports.redundantimport"

Getting such type of error for many different classes.

Thank you

Best Regards,
Deeksha

**Comment by rnveach** [comment]
Provide us the following in raw text, preferably post large chunks of text externally and not to GH. No images except for IDEs or other GUIs (command line is not a GUI), no partials or segmented copies of text.

If you see anything PII in it, feel free to obfuscate it. For example, we shouldn't need to know your computer's user name, but OS and general folder paths might be needed.

1) Confirm you read https://checkstyle.org/beginning_development.html .
2) Confirm you pass all requirements in https://checkstyle.org/beginning_development.html#Before_development .
3) Results of `java -v`.
4) Results of `git status`.
5) Results of `git show-ref origin/master`. `origin` must be switched with the remote that points to the main checkstyle remote, and not your fork or other copy if it isn't already.
6) Description on what you did to get the error. Did you run a command from the command line or through an IDE, and how for either one? We will need to know the actual command and its full results, or the IDE and what you did to get the error in the IDE.
7) Run `mvn clean verify -e` from the command line as the same folder as Checkstyle's POM. Give us the results of the command ONLY if you received a failure, otherwise let us know it passed successfully and how long it took to run. If you were not running this originally on a command line to get the error, we recommend starting with running through command line and only move to the IDE after you did a successful maven run.

I hope to get a clearer picture after this information is provided.

At a glance `imports.redundantimport` is a test location, so I assume you may be getting this error using an IDE.

@romani I recommend posting this a wiki.
